openedx/core/djangoapps/embargo/tests/test_api.py

```diff
@@ -227,10 +228,54 @@ def test_staff_access_country_block(self, staff_role_cls):
 
         # Now the user should have access
         with self._mock_geoip('US'):
-            result = embargo_api.check_course_access(self.course.id, user=self.user, ip_address='0.0.0.0')
+            result = embargo_api.check_course_access(self.course.id, user=self.user, ip_addresses=['0.0.0.0'])
 
         assert result, 'User should have access because the user is staff.'
 
+    @ddt.data(
+        # (Note that any '0.x.x.x' IP _should_ be blocked in this test.)
+        # ips, legacy, allow access
+        (['0.0.0.0', '1.1.1.1'], True, False),  # legacy looks at first IP and blocks
+        (['1.1.1.1', '0.0.0.0'], True, True),  # legacy fails to look at later IPs and allows
+        (['1.1.1.1', '2.2.2.2'], False, True),  # normal chain of access
+        (['1.1.1.1', '0.0.0.0', '2.2.2.2'], False, False),  # tried to sneak a blocked IP in, but we caught it
+    )
+    @ddt.unpack
+    @mock.patch('openedx.core.djangoapps.embargo.api.country_code_from_ip')
+    def test_redirect_if_blocked_ips(self, ips, use_legacy, allow_access, mock_country):
+        # Block the US
+        CountryAccessRule.objects.create(
+            rule_type=CountryAccessRule.BLACKLIST_RULE,
+            restricted_course=self.restricted_course,
+            country=Country.objects.get(country='US')
+        )
+
+        # Treat any ip that starts with zero as from the US
+        mock_country.side_effect = lambda x: 'US' if x.startswith('0.') else 'XX'
+
+        request = RequestFactory().get('', HTTP_X_FORWARDED_FOR=','.join(ips))
+        request.user = self.user
+
+        with override_waffle_switch(USE_LEGACY_IP, use_legacy):
+            assert (embargo_api.redirect_if_blocked(request, self.course.id) is None) == allow_access
+
+    @ddt.data(
+        # access point, check disabled, allow access
+        ('enrollment', False, False),  # 'enrollment' never changes blocked status
+        ('enrollment', True, False),  # 'enrollment' never changes blocked status
+        ('courseware', False, False),  # 'courseware' normally also leaves blocked status in place
+        ('courseware', True, True),  # Unless the access check has been disabled, then we allow them
+    )
+    @ddt.unpack
+    @mock.patch('openedx.core.djangoapps.embargo.api.check_course_access', return_value=False)
+    def test_redirect_if_blocked_courseware(self, access_point, check_disabled, allow_access, _mock_access):
```

Comment:
This `disable_access_check` logic was not changed by my PR, but I noticed there was no test (at this level -- test_middleware.py has some) for it, so added one.